### PR TITLE
Added c64dtv accelerator code and documentation.

### PR DESCRIPTION
**Pull request by blackystardust**
Added c64dtv accelerator code and documentation.

### CODE REVIEW

File: doc/funcref.sgml [hunk 1]
```diff
@@ -2920,6 +2923,26 @@ used in presence of a prototype.
...
+<tag/Header/<tt/<ref id="accelerator.h" name="accelerator.h">/
+<tag/Declaration/<tt/unsigned char detect_c64dtv (void);/
+<tag/Description/The function returns a 1 if a C64DTV has been detected.
+<tag/Notes/<itemize>
+<item>The function is specific to the C64.
+</itemize>
+<tag/Availability/cc65 (not all platforms)
+<tag/See also/
+<ref id="get_scpu_speed" name="get_c64dtv_speed">,
+<ref id="set_scpu_speed" name="set_c64dtv_speed">,
```
Comment (greg-king5) [inline doc/funcref.sgml]: The ID strings, in the references, are wrong.  (A copy-and-paste error from the SCPU functions.)

File: doc/funcref.sgml [hunk 2]
```diff
@@ -3479,6 +3502,28 @@ header files define constants that can be used to check the return code.
...
+
+<quote>
+<descrip>
+<tag/Function/Get the current speed of the C64DTV.
+<tag/Header/<tt/<ref id="accelerator.h" name="accelerator.h">/
+<tag/Declaration/<tt/unsigned char get_c64dtv_speed (void);/
+<tag/Description/The function returns the current speed of the C64DTV.
+<tag/Notes/<itemize>
+<item>The function is specific to the C64.
+<item>The function does not check for the presence of the cartridge.
```
Comment (greg-king5) [inline doc/funcref.sgml]: Not a cartridge.  (Another copy-and-paste error from the SCPU functions.)

File: doc/funcref.sgml [hunk 3]
```diff
@@ -6052,6 +6097,28 @@ clean-up when exitting the program.
 </quote>
 
 
+<sect1>set_c64dtv_speed<label id="set_c64dtv_speed"><p>
+
+<quote>
+<descrip>
+<tag/Function/Set the current speed of the C64DTV.
+<tag/Header/<tt/<ref id="accelerator.h" name="accelerator.h">/
+<tag/Declaration/<tt/unsigned char set_c64dtv_speed (unsigned char speed);/
```
Comment (greg-king5) [inline doc/funcref.sgml]: The `__fastcall__` keyword should be in that line.

File: libsrc/c64/acc_c64dtv_speed.s [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+_set_c64dtv_speed:
+        cmp     #SPEED_2X
+        bcs     high_speed
+low_speed:
+        ldx     #C64DTV_Slow
+set_speed:
+        .byte   $32,$99               ; SAC #$99   set accumulator to reg 9 (cpu control)
+        txa                           ; (re)set skip and burst bits
+        .byte   $32,$00               ; SAC #$99   set accumulator back to reg 0
```
Comment (greg-king5) [inline libsrc/c64/acc_c64dtv_speed.s]: The operand in the comment doesn't match the second byte.

**Comment by blackystardust** [comment]
fixed in my latest commit, I'll need to pay more attention in future pr's, thanx for pointing out the errors.

File: libsrc/c64/acc_c64dtv_speed.s [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+high_speed:
+        ldx     #C64DTV_Fast
+        bne     set_speed
+
+
+_get_c64dtv_speed:
+        .byte   $32,$99               ; SAC #$99   set accumulator to reg 9 (cpu control)
+        tax
+        .byte   $32,$00               ; SAC #$99   set accumulator back to reg 0
```
Comment (greg-king5) [inline libsrc/c64/acc_c64dtv_speed.s]: You forgot to fix that comment.

**Comment by blackystardust** [comment]
Fixed some more comments.

**Comment by blackystardust** [comment]
any more issues or can this one be considered for merging (I have more code waiting to be committed) ??

**Comment by oliverschmidt** [comment]
I see...

**Pull request merged**